Taskfile.yml

```
@@ -0,0 +1,292 @@
+      SHFMT_FILENAME: shfmt_{{.SHFMT_VERSION}}_{{OS}}_{{ARCH}}{{exeExt}}
+      SHFMT_DOWNLOAD_URL: https://github.com/mvdan/sh/releases/download/{{.SHFMT_VERSION}}/{{.SHFMT_FILENAME}}
+    status:
+      - test -x {{.TOOLS_DIR}}/shfmt{{exeExt}}
+    cmds:
+      - curl --fail --silent --show-error --location
+        --url {{.SHFMT_DOWNLOAD_URL}}
+        --output {{.TOOLS_DIR}}/shfmt{{exeExt}}
+      - '{{if ne OS "windows"}}chmod +x {{.TOOLS_DIR}}/shfmt{{end}}'
+
+  # ----------------------------------------------------------------------------
+  # Build tasks
+  # ----------------------------------------------------------------------------
+  clean: rm -rf dist
+
+  gobuild:
+    internal: true
+    generates:
+      - "{{.OUTPUT}}"
+    env:
+      CGO_ENABLED: 0
+      GOOS: "{{.GOOS}}"
+      GOARCH: "{{.GOARCH}}"
+    vars:
+      OUTPUT: '{{joinPath "dist" (list .GOOS .GOARCH | join "-") "bomctl"}}{{.EXT}}'
+    cmd: go build -trimpath -o {{.OUTPUT}} -ldflags="${LDFLAGS}"
+
+  build:
+    desc: Build binaries for all platforms and CPU architectures
+    cmds:
+      - task: build:linux:amd
+      - task: build:linux:arm
+      - task: build:macos:intel
+      - task: build:macos:apple
+      - task: build:windows:amd
+      - task: build:windows:arm
+
+  build:linux:amd:
+    desc: Build for Linux on AMD64
+    cmd:
+      task: gobuild
+      vars:
+        GOOS: linux
+        GOARCH: amd64
+
+  build:linux:arm:
+    desc: Build for Linux on ARM
+    cmd:
+      task: gobuild
+      vars:
+        GOOS: linux
+        GOARCH: arm64
+
+  build:macos:intel:
+    desc: Build for macOS on AMD64
+    cmd:
+      task: gobuild
+      vars:
+        GOOS: darwin
+        GOARCH: amd64
+
+  build:macos:apple:
+    desc: Build for macOS on ARM
+    cmd:
+      task: gobuild
+      vars:
+        GOOS: darwin
+        GOARCH: arm64
+
+  build:windows:amd:
+    desc: Build for Windows on AMD64
+    cmd:
+      task: gobuild
+      vars:
+        EXT: .exe
+        GOOS: windows
+        GOARCH: amd64
+
+  build:windows:arm:
+    desc: Build for Windows on ARM
+    cmd:
+      task: gobuild
+      vars:
+        EXT: .exe
+        GOOS: windows
+        GOARCH: arm64
+
+  # ----------------------------------------------------------------------------
+  # Lint tasks
+  # ----------------------------------------------------------------------------
+  run-lint:
+    silent: true
+    vars:
+      CLI_ARGS: '{{.CLI_ARGS | join " "}}'
+    preconditions:
+      - sh: command -v {{.CLI_NAME}}
+        msg: "{{.CLI_NAME}} not found, please install and run the command again."
+    cmds:
+      - printf "\nRunning ${CYAN}{{.CLI_NAME}} {{.CLI_ARGS}}${RESET}\n"
+      - PATH={{.TOOLS_DIR}}{{eq OS "windows" | ternary ";" ":"}}${PATH} {{.CLI_NAME}} {{.CLI_ARGS}}
+
+  lint:
+    desc: Lint Golang code, markdown, shell scripts, and YAML files
+    cmds:
+      - task: lint:go
+      - task: lint:markdown
+      - task: lint:shell
+      - task: lint:yaml
```

```suggestion
  lint:
    desc: Lint Golang code, markdown, shell scripts, and YAML files
    cmds:
      - task: lint:go
      - task: lint:markdown
      - task: lint:shell
      - task: lint:yaml
      
   lint-fix:
    desc: Lint Golang code, markdown, shell scripts, and YAML files and fix where possible
    cmds:
      - task: lint:go:fix
      - task: lint:markdown:fix
      - task: lint:shell
      - task: lint:yaml
```

To have parity with the makefile

---

Ok I'll make that change, good call. It will need to be `lint:fix` instead of `lint-fix` though, I think I'll add a `lint:shell:fix` as well to let `shfmt` fix shell scripts as well as just checking